public/app/features/explore/Explore.tsx

```diff
@@ -219,15 +220,29 @@ export class Explore extends React.PureComponent<Props, ExploreState> {
   /**
    * Used by Logs details.
    */
-  onClickFilterLabel = (key: string, value: string, refId?: string) => {
-    this.onModifyQueries({ type: 'ADD_FILTER', options: { key, value } }, refId);
+  onClickFilterLabel = (key: string, value: string, refId?: string, frame?: DataFrame) => {
```

Comment:
We have the `refId` in the frame. Can we use it from frame and simplify this?

Reply:
Good call, done in 1b9eaf6a357da71eb1cc178a32eacf4c48ecbdf8